scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala

```diff
@@ -1416,4 +1465,47 @@ object FormatWriter {
 
   private val leadingPipeSpace = compileStripMarginPattern('|')
 
+  /** [[https://dotty.epfl.ch/docs/reference/other-new-features/indentation.html#the-end-marker]] */
+  private def getEndMarkerLabel(tree: Tree): String = tree match {
+    // templates
+    case _: Term.NewAnonymous => "new"
+    case t: Defn.Class => t.name.value
+    case t: Defn.Object => t.name.value
+    case t: Defn.Trait => t.name.value
+    case t: Defn.Enum => t.name.value
+    case t: Defn.Given => t.name.value
+    case t: Pkg.Object => t.name.value
+    // definitions
+    case t: Defn.Def => t.name.value
+    case t: Defn.Macro => t.name.value
```

Comment:
This is not used in Scala 3, so I would remov eit.

Reply:
done